util/src/main/java/com/coderanch/util/cli/InputUtility.java

```diff
@@ -0,0 +1,267 @@
+    private OptionalInt tryIntParse(String line, Predicate<? super Integer> intPredicate) {
+        try {
+            var num = Integer.parseInt(line);
+            return intPredicate.test(num) ? OptionalInt.of(num) : OptionalInt.empty();
+        }
+        catch (NumberFormatException e) {
+            return OptionalInt.empty();
+        }
+    }
+
+    /**
+     * Displays the prompt and reads the input stream, returning a validated real number.
+     *
+     * @param prompt          the prompt to display to the user.
+     * @param doublePredicate the predicate to use for validation.
+     * @return a validated floating point number.
+     * @throws IOException when there's a problem with {@link InputStream}
+     */
+    public double nextDouble(String prompt, Predicate<? super Double> doublePredicate) throws IOException {
+        System.out.println(prompt);
+        String line;
+        OptionalDouble num;
+        for (
+            line = reader.readLine(), num = OptionalDouble.empty();
+            num.isEmpty();
+            line = reader.readLine()
+        ) {
+            num = tryDoubleParse(line, doublePredicate);
+            if (num.isEmpty()) {
+                System.out.println("Invalid input.");
+            }
+        }
+        return num.getAsDouble();
+    }
+
+    /**
+     * Tries to parse a Double from the string and validate it.
+     *
+     * @param line            the string to parse.
+     * @param doublePredicate the predicate used to validate the Integer.
+     * @return a validated Double or nothing.
+     */
+    private OptionalDouble tryDoubleParse(String line, Predicate<? super Double> doublePredicate) {
+        try {
+            var num = Double.parseDouble(line);
+            return doublePredicate.test(num) ? OptionalDouble.of(num) : OptionalDouble.empty();
+        }
+        catch (NumberFormatException e) {
+            return OptionalDouble.empty();
+        }
+    }
+
+    /**
+     * Gets the next yes or no answer.
+     * Calls {@link InputUtility#nextString(String, Predicate)}.
+     * Passes the prompt and a predicate that validates "y", "n", "yes", "no".
+     *
+     * @param prompt the prompt to display to the user.
+     * @return true for yes and false for no.
+     * @throws IOException when there's a problem with {@link InputStream}
+     */
+    public boolean nextYesNo(String prompt) throws IOException {
+        var result = this.nextString(prompt, yesOrNo()).trim().toLowerCase();
+        return YES_SYNONYMS.contains(result);
+    }
+
+    /**
+     * Calls {@link InputUtility#pause(String)}.
+     * The prompt defaults to "Press when ready".
+     *
+     * @throws IOException when there's a problem with {@link InputStream}
+     */
+    public void pause() throws IOException {
+        pause("Press when ready");
+    }
+
+    /**
+     * Calls nextString{@link InputUtility#nextString(String, Predicate)}.
+     * Uses prompt entered and waits for the user to press <enter>.
```

Comment:
Remember, Javadoc includes HTML syntax, so you need to escape special characters.
```suggestion
     * Uses prompt entered and waits for the user to press &lt;enter&gt;.
```